### PR TITLE
click on nicks to insert them into the MessageComposer

### PR DESCRIPTION
**Pull request by ara4n**

### CODE REVIEW

**Comment by ara4n** [comment]
oops, i missed someone; one sec

**Comment by ara4n** [comment]
right, focus sorted now - this pairs nicely with a glass of fine 1963 https://github.com/vector-im/vector-web/pull/1191

File: src/components/views/rooms/EventTile.js [hunk 1]
```diff
@@ -318,12 +325,12 @@ module.exports = React.createClass({
...
                     <div className="mx_EventTile_avatar">
                         <MemberAvatar member={this.props.mxEvent.sender} width={24} height={24}
-                         onClick={ this.onMemberAvatarClicked.bind(this, this.props.mxEvent.sender) } />
+                         onClick={ this.onMemberAvatarClick } />
                     </div>
                 );
             }
             if (EventTileType.needsSenderProfile()) {
-                sender = <SenderProfile mxEvent={this.props.mxEvent} aux={aux} />;
+                sender = <SenderProfile onClick={ this.onSenderProfileClick } mxEvent={this.props.mxEvent} aux={aux} />;
```
Comment (richvdh) [inline src/components/views/rooms/EventTile.js]: is there an associated vector-web change to add an onClick property to SenderProfile?

File: src/components/views/rooms/MessageComposer.js [hunk 2]
```diff
@@ -192,9 +192,29 @@ module.exports = React.createClass({
...
+                    }
+                    else {
+                        left = left.replace(/( ?)$/, " " + payload.displayname);
+                    }
+                    textarea.value = left + right;
+                    textarea.focus();
+                    textarea.setSelectionRange(left.length, left.length);
+                }
+                else {
+                    this.refs.textarea.value = payload.displayname + ": ";
```
Comment (richvdh) [inline src/components/views/rooms/MessageComposer.js]: s/this.refs.// here and below ?

File: src/components/views/rooms/EventTile.js (hunk 1 above)
Comment (ara4n) [inline src/components/views/rooms/EventTile.js]: yup, see main PR comment stream

File: src/components/views/rooms/MessageComposer.js [hunk 3]
```diff
@@ -192,9 +192,29 @@ module.exports = React.createClass({
...
         switch (payload.action) {
             case 'focus_composer':
-                this.refs.textarea.focus();
+                textarea.focus();
+                break;
+            case 'insert_displayname':
+                if (textarea.value) {
+                    var left = textarea.value.substring(0, textarea.selectionStart);
+                    var right = textarea.value.substring(textarea.selectionEnd);
+                    if (right && right.length) {
```
Comment (richvdh) [inline src/components/views/rooms/MessageComposer.js]: what is the null-guard here for? or, maybe more pertinently: why have you decided you need one here, but not anywhere else?

File: src/components/views/rooms/MessageComposer.js (hunk 2 above)
Comment (ara4n) [inline src/components/views/rooms/MessageComposer.js]: fixed, thanks

File: src/components/views/rooms/MessageComposer.js (hunk 3 above)
Comment (ara4n) [inline src/components/views/rooms/MessageComposer.js]: `left` will always have a value (even if it's a null string), as it's the substring from index 0 to a string of known finite length.  this is the only place we dereference `right`.  Ideally we'd be put checks everywhere but it'd clutter it up quite a lot...

**Comment by ara4n** [comment]
@richvdh PTAL

File: src/components/views/rooms/MessageComposer.js (hunk 3 above)
Comment (ara4n) [inline src/components/views/rooms/MessageComposer.js]: actually, we don't need a nullguard on right as "asd".substring(10) returns '' - my bad.

Comment (richvdh) [inline src/components/views/rooms/MessageComposer.js]: ftr, I was worried about `textarea.value = left + right` which does the wrong thing if right is undefined/null. But it seems to be moot

**Comment by richvdh** [comment]
lgtm

**Pull request merged**